lms/djangoapps/discussion_api/tests/test_permissions.py

```diff
@@ -2,37 +2,86 @@
 Tests for discussion API permission logic
 """
 import itertools
-from unittest import TestCase
 
 import ddt
 
-from discussion_api.permissions import can_delete, get_editable_fields
+from discussion_api.permissions import (
+    can_delete,
+    get_editable_fields,
+    get_initializable_comment_fields,
+    get_initializable_thread_fields,
+)
 from lms.lib.comment_client.comment import Comment
 from lms.lib.comment_client.thread import Thread
 from lms.lib.comment_client.user import User
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory
 
 
-def _get_context(requester_id, is_requester_privileged, thread=None):
+def _get_context(requester_id, is_requester_privileged, is_cohorted=False, thread=None):
     """Return a context suitable for testing the permissions module"""
     return {
         "cc_requester": User(id=requester_id),
         "is_requester_privileged": is_requester_privileged,
+        "course": CourseFactory(cohort_config={"cohorted": is_cohorted}),
         "thread": thread,
     }
 
 
 @ddt.ddt
-class GetEditableFieldsTest(TestCase):
-    """Tests for get_editable_fields"""
+class GetInitializableFieldsTest(ModuleStoreTestCase):
+    """Tests for get_*_initializable_fields"""
```

Comment:
Style nit: one-line docstrings should have opening and closing quotes on their own lines.